services/migrations/dump.go

```diff
@@ -635,3 +635,34 @@ func RestoreRepository(ctx context.Context, baseDir, ownerName, repoName string,
 	}
 	return updateMigrationPosterIDByGitService(ctx, structs.GitServiceType(tp))
 }
+
+// RestoreFromGithubExportedData restore a repository from the disk directory
+func RestoreFromGithubExportedData(ctx context.Context, baseDir, ownerName, repoName string, units []string) error {
+	doer, err := user_model.GetAdminUser()
+	if err != nil {
+		return err
+	}
+	var uploader = NewGiteaLocalUploader(ctx, doer, ownerName, repoName)
+	downloader, err := NewRepositoryRestorer(ctx, baseDir, ownerName, repoName)
```

Comment:
I'm sure I'm misunderstanding the control flow here, but shouldn't this be
```suggestion
	downloader, err := NewGithubRepositoryRestorer(ctx, baseDir, ownerName, repoName)
```